mantidimaging/gui/windows/spectrum_viewer/presenter.py

```
@@ -318,6 +324,8 @@ def do_add_roi(self) -> None:
         Add a new ROI to the spectrum
         """
         roi_name = self.model.roi_name_generator()
+        if roi_name in self.view.spectrum_widget.roi_dict:
```

Can we add a test to check we can raise this exception?